src/utils/TraceUtils.js

```diff
@@ -177,3 +170,73 @@ export const treeDecorate = (key, f, tree) => {
     newTree[key] = f(tree);
     return newTree;
 }
+
+export const highlightProperty = 'isHighlighted';
+
+// Takes a tree and attaches the property 'isHighlighted' to every node in the
+// tree. All those nodes whose trace has the given trace as prefix will get
+// 'isHiglighted' = true, and the rest false.
+export const forwardMatching = (trace, tree) => {
+    let f;
+    if (trace === null) {
+        // Completely null traces do not match anything.
+        f = (t) => (false);
```

Comment:
Actually, neither is the parentheses around `t`, I learned. But I like to have them anyway. But willr emove around false, for consitency.